### PR TITLE
Update Gutenberg lifecycle clarity and grammar

### PR DESCRIPTION
**Pull request by rjcdr**
## Description
Simplified sentence structure, semantics of the closing paragraph. Also made redundant content relevant to the overall aims of the passage.

### CODE REVIEW

File: docs/designers-developers/key-concepts.md [hunk 1]
```diff
@@ -143,4 +143,4 @@ A purely dynamic block that is to be server rendered before display could look l
...
+In summary, the workflow for editing a Gutenberg post starts with using token delimeters to generate an in-memory tree from the persisted version of the document. The process ends by reversing the serialization of blocks into the `post_content`. During editing, all manipulations happen within the block tree. Alternatively, a Gutenberg post is built and manipulated using an isomorphically persistied, in-memory data structure. Although Gutenberg post persistence is currently achieved using a serialization/parser pair, in hypothetical scenarios the post data structure could also be stored using a plugin or retreived from a remote JSON file.
```
Comment (aduth) [inline docs/designers-developers/key-concepts.md]: >The process ends by reversing the serialization of blocks

The phrasing here seems to imply that a serialization is reversed, which doesn't really make sense. The previous phrasing was more to emphasize the _inverse_ of the initial parse, i.e. completing the cycle of "Markup -> (Parse) -> Blocks -> (Serialize) -> Markup".

**Comment by mkaz** [comment]
My two cents both versions are fairly dense and difficult to read, here is a potential rewrite for simplicity

> In summary, the Gutenberg workflow parses the saved document to an in-memory tree of blocks, using token delimiters to help. During editing, all manipulations happen within the block tree. The process ends by serializing the blocks back to the `post_content`. 

> The workflow process relies on a serialization/parser pair to persist posts. Hypothetically, the post data structure could be stored using a plugin or retrieved from a remote JSON file to be converted to the block tree.

**Comment by gziolo** [comment]
I applied changes proposed by @mkaz - is it good to go now?

**Review by mkaz — APPROVED** [review]
Looks good to me ;-)

**Pull request merged**